### PR TITLE
[service.subtitles.pipocas] 1.0.1 Matrix ready

### PR DESCRIPTION
**Pull request by highlandr**
### Description
Code remake for Matrix compatibility.

### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

### CODE REVIEW

**Comment by highlandr** [comment]
@enen92 Here we go again :) Check if all is ok.
Thanks.

Happy new year!

File: service.subtitles.pipocas/addon.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - v1.0.1 (2022-01-04) - Matrix READY!
+    </news>
+    <platform>all</platform>
+    <assets>
+      <icon>icon.png</icon>
+    </assets>
+    <license>GPL-2.0-only</license>
+    <forum>https://forum.kodi.tv/forumdisplay.php?fid=143</forum>
+    <website>https://www.pipocas.tv</website>
+    <email></email>
```
Comment (enen92) [inline service.subtitles.pipocas/addon.xml:27]: please remove email tag if empty

File: service.subtitles.pipocas/resources/lib/pipocas.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import sys
+import time
+import unicodedata
+import urllib
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcvfs
+import fnmatch
+try:
```
Comment (enen92) [inline service.subtitles.pipocas/resources/lib/pipocas.py:18]: please remove simplejson usage, from Helix and up you can simply `import json`

Comment (enen92) [reply service.subtitles.pipocas/resources/lib/pipocas.py:18]: you don't even import it as a dependency so it doesn't make much sense, will always throw.

File: service.subtitles.pipocas/service.py [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+        content = sessionPipocasTv.get(url)
+
+    # Bubble sort, to put syncs on top
+    subtitles_list = bubbleSort(subtitles_list)
+    return subtitles_list
+
+
+def append_subtitle(item):
+    
+    listitem = xbmcgui.ListItem(
```
Comment (enen92) [inline service.subtitles.pipocas/service.py:241]: in all your listitems please use offscreen=True. Reason: https://xbmc.github.io/docs.kodi.tv/master/kodi-base/d8/d29/group__python__xbmcgui__listitem.html#ga15bcb930a4f91dd97f8ed7c633a20618

File: service.subtitles.pipocas/service.py [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+    url = INTERNAL_LINK_URL % args
+    ## add it to list, this can be done as many times as needed for all subtitles found
+    xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=listitem, isFolder=False)
+
+class Main:
+
+
+    def Search(item):
+        log("Host data '%s'" % OS_DETECT)
+        """Called when searching for subtitles from XBMC."""
```
Comment (enen92) [inline service.subtitles.pipocas/service.py:262]:
```suggestion
        """Called when searching for subtitles from Kodi."""
```

:)

File: service.subtitles.pipocas/service.py [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+        # we pickup all our arguments sent from def Search()
+        subs = Download(params["id"], params["filename"])
+
+        # we can return more than one subtitle for multi CD versions, for now we
+        # are still working out how to handle that in XBMC core
+        for sub in subs:
+            listitem = xbmcgui.ListItem(label=sub)
+            xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=sub, listitem=listitem, isFolder=False)
+
+    # Send end of directory to XBMC
```
Comment (enen92) [inline service.subtitles.pipocas/service.py:603]:
```suggestion
    # Send end of directory to Kodi
```

File: service.subtitles.pipocas/default.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# Service Pipocas.tv
+
+#simple call, it's just a subtitle addon
+from service import Main
+
+#run the program
+Main().get_params
```
Comment (enen92) [inline service.subtitles.pipocas/default.py:8]: are you sure this works? `get_params` is not annotated as a `@property` so I doubt you can run the method like this. Missing ()?

```suggestion
Main().get_params()
```

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments

File: service.subtitles.pipocas/default.py (hunk 6 above)
Comment (highlandr) [reply service.subtitles.pipocas/default.py:8]: It does work :) tested but I'll change it.

File: service.subtitles.pipocas/addon.xml (hunk 1 above)
Comment (highlandr) [reply service.subtitles.pipocas/addon.xml]: done.

File: service.subtitles.pipocas/resources/lib/pipocas.py (hunk 2 above)
Comment (highlandr) [reply service.subtitles.pipocas/resources/lib/pipocas.py]: done.

File: service.subtitles.pipocas/service.py (hunk 3 above)
Comment (highlandr) [reply service.subtitles.pipocas/service.py:241]: done

File: service.subtitles.pipocas/service.py (hunk 4 above)
Comment (highlandr) [reply service.subtitles.pipocas/service.py]: done

File: service.subtitles.pipocas/service.py (hunk 5 above)
Comment (highlandr) [reply service.subtitles.pipocas/service.py]: done

File: service.subtitles.pipocas/default.py (hunk 6 above)
Comment (highlandr) [reply service.subtitles.pipocas/default.py:8]: done

**Comment by highlandr** [comment]
@enen92 should be good to go :)
Thanks!

**Review by enen92 — APPROVED** [review]

**Pull request merged**